src/Sulu/Bundle/AdminBundle/Resources/js/containers/ViewRenderer/ViewRenderer.js

```diff
@@ -4,22 +4,35 @@ import type {Element} from 'react';
 import {observer} from 'mobx-react';
 import {sidebarStore} from '../Sidebar';
 import Router from '../../services/Router';
-import type {Route} from '../../services/Router';
+import type {AttributeMap, Route} from '../../services/Router';
 import viewRegistry from './registries/ViewRegistry';
 import type {View} from './types';
 
 type Props = {
     router: Router,
 };
 
+const UPDATE_ROUTE_HOOK_PRIORITY = -1024;
+
 @observer
 class ViewRenderer extends React.Component<Props> {
     componentDidUpdate() {
         this.clearSidebarConfig();
     }
 
     componentDidMount() {
+        const {router} = this.props;
+
         this.clearSidebarConfig();
+
+        router.addUpdateRouteHook((newRoute, newAttributes) => {
+            const {attributes: oldAttributes, route: oldRoute} = router;
+            if (this.getKey(newRoute, newAttributes) !== this.getKey(oldRoute, oldAttributes)) {
+                router.clearBindings();
```

Comment:
Previously was forgotten to clear the bindings on the router on the `restore` operation, but it was correctly done on the `navigate` and `redirect` operations.

At this place it is done for all three of them, so that an error like this should not occur anymore. The other reason to move it here is another bug I've encountered: If one of the `updateRouteHooks` registered cancel the navigation, the bindings were already cleared, although the navigation did not happen at all, which should also be fixed with this implementation. (However, this does not happen in the core currently, but can be tested by adding a `updateRouteHook` returning false anywhere.

Reply:
👍